crates/python/src/grpc/models/controller.rs

```diff
@@ -14,6 +15,27 @@ py_wrap_data_struct! {
     }
 }
 
+#[pymethods]
+impl PyReadoutValues {
+    pub fn as_ndarray(&self, py: Python<'_>) -> PyResult<Option<PyObject>> {
+        match &self.as_inner().values {
+            None => Ok(None),
+            Some(Values::IntegerValues(ints)) => {
+                Ok(Some(PyArray::from_slice(py, &ints.values).to_object(py)))
+            }
+            Some(Values::ComplexValues(complex)) => Ok(Some(
```

Comment:
Why does `PyReadoutValues` only support integer or complex numbers, while the raw version supports reals as well? Do we just return reals as complex numbers with a real-part of 0?

Reply:
Ah, there are two `PyReadoutValues`. This one is a raw mapping to the controller service gRPC model of readout values, which only supports `int` and `complex`. I actually don't think this method is even needed any more, as I'm using the higer level `PyReadoutValues` that supports `real` as well. The only reason for the difference between the model and the SDK version is for future proofing. If we ever return real numbered data from the QPU, we don't have to make a breaking change.